File: PontosColeta.WebApp/Scripts/app/places/PlaceWorkingDay.ts

export class PlaceWorkingDay {
    private static daysOfWeek = ["Domingo", "Segunda", "Terça", "Quarta", "Quinta", "Sexta", "Sábado"];
    constructor(public DayOfWeek: number) { }

    IsEnabled = false;

    StartTime: string;

    EndTime: string;

    get Name() {
        return PlaceWorkingDay.daysOfWeek[this.DayOfWeek];
    }

}

File: PontosColeta.UserApp/PontosColeta.UserApp/Models/PlaceWorkingDay.cs
using System;
using System.Collections.Generic;

using System.Linq;
using System.Web;

namespace PontosColeta.UserApp
{
    public class PlaceWorkingDay
    {
        public int Id { get; set; }

        public DayOfWeek DayOfWeek { get; set; }

        public TimeSpan? StartTime { get; set; }

        public TimeSpan? EndTime { get; set; }
        
        public bool IsEnabled => Id != 0;
    }
}